### PR TITLE
Tests/android UI tests absolute

### PR DESCRIPTION
**Pull request by stefanosiano**
## :scroll: Description
We have benchmarks that calculate relative difference between two operations with percentages.
We are now adding absolute difference difference between two operations.

## :bulb: Motivation and Context
We want to make sure that certain operations don't take too long. E.g. the sdk init shouldn't take more than a few milliseconds.

#skip-changelog

## :green_heart: How did you test it?
🤣 

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] No breaking changes

## :crystal_ball: Next steps
Check other devices we can run the benchmarks on (there is still some issues with android runner on old versions). Will add them in another pr.

### CODE REVIEW

**Comment by marandaneto** [comment]
@stefanosiano let us know once this PR is ready to be reviewed since there are still commits going on, thanks.

**Comment by stefanosiano** [comment]
> @stefanosiano let us know once this PR is ready to be reviewed since there are still commits going on, thanks.

This pr is ready. I'm just making a few changes for the new github actions, but only on the .yml files

File: .github/workflows/tests-benchmark.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: "Integration Tests"
```
Comment (marandaneto) [inline .github/workflows/tests-benchmark.yml:1]: I'd rather rename the `name` or the file.
name is integration tests and the file is tests benchmark, which one is the correct one?

File: .github/workflows/tests-benchmark.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+jobs:
+  build:
+    name: Benchmarks
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        # Zulu Community distribution of OpenJDK
+        java: ["11"]
```
Comment (marandaneto) [inline .github/workflows/tests-benchmark.yml:17]: If we are running only on ubuntu and Java 11, we don't need a matrix.

File: .github/workflows/tests-uitests.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        run: make stop
+
+      - name: Install SauceLab command line
+        run: npm install -g saucectl
+
+      - name: Run Tests in SauceLab
+        env:
+          SAUCE_USERNAME: ${{ secrets.SAUCE_USERNAME }}
+          SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
+        run: saucectl run -c .sauce/sentry-uitest-android-end2end.yml
```
Comment (marandaneto) [inline .github/workflows/tests-uitests.yml:45]: Can we actually use https://github.com/marketplace/actions/saucectl-run-action ?

File: sentry-android-integration-tests/sentry-uitest-android-benchmark/src/androidTest/java/io/sentry/uitest/android/benchmark/SdkBenchmarkTest.kt [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            SentryAndroid.init(context) {
+                it.dsn = "https://key@host/proj"
+            }
+        }
+        val opNoSdk2 = getOperation()
+        val opPerfProfilingSdk = getOperation {
+            SentryAndroid.init(context) {
+                it.dsn = "https://key@host/proj"
+                it.isProfilingEnabled = true
+                it.isEnableAutoActivityLifecycleTracing = true
```
Comment (marandaneto) [inline sentry-android-integration-tests/sentry-uitest-android-benchmark/src/androidTest/java/io/sentry/uitest/android/benchmark/SdkBenchmarkTest.kt:53]: This is already enabled by default.

File: sentry-android-integration-tests/sentry-uitest-android-benchmark/src/androidTest/java/io/sentry/uitest/android/benchmark/SentryBenchmarkTest.kt [hunk 5]
```diff
@@ -53,35 +45,42 @@ class SentryBenchmarkTest {
...
         val benchmarkOperationProfiled = BenchmarkOperation(
             choreographer,
-            getOperation(runner) {
+            before = {
+                runner.runOnMainSync {
+                    SentryAndroid.init(context) { options: SentryOptions ->
+                        options.dsn = "https://key@uri/1234567"
+                        options.tracesSampleRate = 1.0
+                        options.isProfilingEnabled = true
+                        options.isEnableAutoSessionTracking = false
```
Comment (marandaneto) [inline sentry-android-integration-tests/sentry-uitest-android-benchmark/src/androidTest/java/io/sentry/uitest/android/benchmark/SentryBenchmarkTest.kt:73]: Any reason to disable `isEnableAutoSessionTracking`? just out of curiosity

File: sentry-android-integration-tests/sentry-uitest-android-benchmark/src/androidTest/java/io/sentry/uitest/android/benchmark/util/BenchmarkOperation.kt [hunk 6]
```diff
@@ -33,6 +38,9 @@ internal class BenchmarkOperation(private val choreographer: Choreographer, priv
             warmupIterations: Int = 3,
             measuredIterations: Int = 15
         ): BenchmarkResult {
+            // Android pushes the "installed app" event to othrt apps and the system itself.
```
Comment (marandaneto) [inline sentry-android-integration-tests/sentry-uitest-android-benchmark/src/androidTest/java/io/sentry/uitest/android/benchmark/util/BenchmarkOperation.kt:41]:
```suggestion
            // Android pushes the "installed app" event to other apps and the system itself.
```

**Comment by stefanosiano** [comment]
fixed the tests and now we have them running on SauceLabs!
I had to remove the dependency between `sentry-uitest-android-benchmark` and `sentry-uitest-android`.
Also `sentry-uitest-android` module was moved from `com.android.library` to `com.android.application`, so that we can build an apk to upload to Saucelabs

File: .sauce/sentry-uitest-android-benchmark.yml [hunk 7]
```diff
@@ -2,25 +2,59 @@ apiVersion: v1alpha
...
+      useTestOrchestrator: true
+
+  - name: "Android 11 (api 30)"
+    devices:
+      - id: OnePlus_9_Pro_real_us # OnePlus 9 Pro - api 30 (11)
+      - id: Google_Pixel_2_real_us # Google Pixel 2 - api 30 (11)
+    testOptions:
+      useTestOrchestrator: true
+
+# Commenting for the moment, due to the error "Cannot install test-services-1.4.1.apk on device" on low Android versions
```
Comment (marandaneto) [inline .sauce/sentry-uitest-android-benchmark.yml:36]: Why do we get this error? Should we fix or remove the comments?

Comment (stefanosiano) [reply .sauce/sentry-uitest-android-benchmark.yml:36]: The errors are related to an issue with Android test orchestrator on old versions.
I will fix the issue, but in another pr, so we can understand how the sauce labs runs work with the other commits (adding devices would make the tests take longer)

**Review by marandaneto — APPROVED** [review]
LGTM

**Pull request merged**